gson/src/main/java/com/google/gson/ReflectionAccessFilter.java

```diff
@@ -0,0 +1,187 @@
+   * returns {@link FilterResult#INDECISIVE}.
+   *
+   * <p>This filter is mainly intended to help enforcing the access checks of
+   * Java Platform Module System. It allows detecting illegal access, even if
+   * the used Java version would only log a warning, or is configured to open
+   * packages for reflection. However, this filter <b>only works for Java 9 and
+   * higher</b>, when using an older Java version its functionality will be
+   * limited.
+   *
+   * <p>Note that this filter might not cover all standard Java classes. The set
+   * of detected classes might be expanded in the future without prior notice.
+   *
+   * @see FilterResult#BLOCK_INACCESSIBLE
+   */
+  ReflectionAccessFilter BLOCK_INACCESSIBLE_JAVA = new ReflectionAccessFilter() {
+    @Override public FilterResult check(Class<?> rawClass) {
+      return ReflectionAccessFilterHelper.isJavaType(rawClass)
+        ? FilterResult.BLOCK_INACCESSIBLE
+        : FilterResult.INDECISIVE;
+    }
+  };
+
+  /**
+   * Blocks all reflection access to members of standard Java classes.
+   *
+   * <p>If this filter encounters a class other than a standard Java class it
+   * returns {@link FilterResult#INDECISIVE}.
+   *
+   * <p>This filter is mainly intended to prevent depending on implementation
+   * details of the Java platform and to help applications prepare for upgrading
+   * to the Java Platform Module System.
+   *
+   * <p>Note that this filter might not cover all standard Java classes. The set
+   * of detected classes might be expanded in the future without prior notice.
+   *
+   * @see #BLOCK_INACCESSIBLE_JAVA
+   * @see FilterResult#BLOCK_ALL
+   */
+  ReflectionAccessFilter BLOCK_ALL_JAVA = new ReflectionAccessFilter() {
+    @Override public FilterResult check(Class<?> rawClass) {
+      return ReflectionAccessFilterHelper.isJavaType(rawClass)
+        ? FilterResult.BLOCK_ALL
+        : FilterResult.INDECISIVE;
+    }
+  };
+
+  /**
+   * Blocks all reflection access to members of standard Android classes.
+   *
+   * <p>If this filter encounters a class other than a standard Android class it
+   * returns {@link FilterResult#INDECISIVE}.
+   *
+   * <p>This filter is mainly intended to prevent depending on implementation
+   * details of the Android platform.
+   *
+   * <p>Note that this filter might not cover all standard Android classes. The set
+   * of detected classes might be expanded in the future without prior notice.
+   *
+   * @see FilterResult#BLOCK_ALL
+   */
+  ReflectionAccessFilter BLOCK_ALL_ANDROID = new ReflectionAccessFilter() {
+    @Override public FilterResult check(Class<?> rawClass) {
+      return ReflectionAccessFilterHelper.isAndroidType(rawClass)
+        ? FilterResult.BLOCK_ALL
+        : FilterResult.INDECISIVE;
+    }
+  };
+
+  /**
+   * Blocks all reflection access to members of classes belonging to programming
+   * language platforms, such as Java, Android, Kotlin or Scala.
```

Comment:
"This includes all classes that would be blocked by {@link #BLOCK_ALL_JAVA} or {@link #BLOCK_ALL_ANDROID}, as well as language-specific platform classes like {@code kotlin.*}." I think we can avoid spelling out exactly what those language-specific classes are.